In [1]:
# Necessary imports
import pandas as pd
import numpy as np

In [2]:
# Reading DataFrame
data = pd.read_csv('df_sex_unemployment_rates.csv')

In [3]:
# Dataframe overview
data

date  overall_rate  men_rate  women_rate  men_16_17_rate  \
0    1948-01-01           3.4       3.4         3.3             9.7   
1    1948-02-01           3.8       3.6         4.5            13.0   
2    1948-03-01           4.0       3.8         4.4            14.0   
3    1948-04-01           3.9       3.8         4.3            11.6   
4    1948-05-01           3.5       3.5         3.7             7.1   
..          ...           ...       ...         ...             ...   
902  2023-03-01           3.5       3.6         3.4             9.5   
903  2023-04-01           3.4       3.5         3.3             8.6   
904  2023-05-01           3.7       3.7         3.6             8.0   
905  2023-06-01           3.6       3.7         3.4            14.1   
906  2023-07-01           3.5       3.6         3.4            13.2   

     women_16_17_rate  men_16_19_rate  women_16_19_rate  men_18_19_rate  \
0                 8.8             9.4               7.2             9.5   
1                13.2            10.8               8.9             9.2   
2                11.4            11.9               8.6            10.3   
3                10.6             9.8               9.2             8.6   
4                 5.4             7.6               6.1             8.6   
..                ...             ...               ...             ...   
902               8.7            10.4               9.3            10.6   
903              10.2             9.1               9.2            10.1   
904              10.7            10.4              10.2            11.5   
905               8.7            12.5               9.7            10.5   
906               9.4            13.0               9.7            12.9   

     women_18_19_rate  ...  men_25_34_rate  women_25_34_rate  men_25_54_rate  \
0                 6.8  ...             2.6               4.3             2.3   
1                 6.8  ...             2.7               5.1             2.6   
2                 7.3  ...             2.7               3.5             2.6   
3                 8.6  ...             3.2               3.8             2.8   
4                 7.0  ...             2.9               3.3             2.5   
..                ...  ...             ...               ...             ...   
902               9.9  ...             4.0               3.4             3.0   
903               8.7  ...             4.0               3.5             3.1   
904               9.5  ...             3.7               4.0             3.1   
905              10.4  ...             4.0               3.9             3.0   
906              10.0  ...             3.8               3.3             3.1   

     women_25_54_rate  men_35_44_rate  women_35_44_rate  men_45_54_rate  \
0                 2.8             2.1               1.8             2.3   
1                 3.7             2.5               2.6             2.6   
2                 3.3             2.6               3.0             2.4   
3                 3.5             2.7               3.5             2.5   
4                 3.1             2.4               3.0             2.3   
..                ...             ...               ...             ...   
902               2.8             2.8               2.3             2.1   
903               3.0             2.9               3.0             2.4   
904               3.2             2.7               3.1             2.7   
905               3.2             2.5               3.5             2.5   
906               2.8             2.8               3.2             2.6   

     women_45_54_rate  men_55plus_rate  women_55plus_rate  
0                 2.1              3.0                NaN  
1                 3.3              2.9                NaN  
2                 3.3              2.8                NaN  
3                 3.1              2.9                NaN  
4                 2.9              3.1                NaN  
..                ...              

This dataframe contains 907 entries and 26 variables.

The collection began in 01/01/1948 and end in 07/01/2023.

In [4]:
# Checking all variables available
data.columns

Index(['date', 'overall_rate', 'men_rate', 'women_rate', 'men_16_17_rate',
       'women_16_17_rate', 'men_16_19_rate', 'women_16_19_rate',
       'men_18_19_rate', 'women_18_19_rate', 'men_16_24_rate',
       'women_16_24_rate', 'men_20_24_rate', 'women_20_24_rate',
       'men_25plus_rate', 'women_25plus_rate', 'men_25_34_rate',
       'women_25_34_rate', 'men_25_54_rate', 'women_25_54_rate',
       'men_35_44_rate', 'women_35_44_rate', 'men_45_54_rate',
       'women_45_54_rate', 'men_55plus_rate', 'women_55plus_rate'],
      dtype='object')

There are many ranges that are overlapping and can be uneccesary for our analysis

In [5]:
# All variables that can be discarded
cols_drop = ['men_16_17_rate',
       'women_16_17_rate','men_18_19_rate', 'women_18_19_rate', 'men_16_24_rate',
       'women_16_24_rate','men_25plus_rate', 'women_25plus_rate', 'men_25_54_rate', 'women_25_54_rate']

# Dropping variables we want to discard
data.drop(columns=cols_drop, inplace = True)


In [6]:
# Checking data non null value counts and variable data types
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 907 entries, 0 to 906
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   date               907 non-null    object 
 1   overall_rate       907 non-null    float64
 2   men_rate           907 non-null    float64
 3   women_rate         907 non-null    float64
 4   men_16_19_rate     907 non-null    float64
 5   women_16_19_rate   907 non-null    float64
 6   men_20_24_rate     907 non-null    float64
 7   women_20_24_rate   907 non-null    float64
 8   men_25_34_rate     907 non-null    float64
 9   women_25_34_rate   907 non-null    float64
 10  men_35_44_rate     907 non-null    float64
 11  women_35_44_rate   907 non-null    float64
 12  men_45_54_rate     907 non-null    float64
 13  women_45_54_rate   907 non-null    float64
 14  men_55plus_rate    907 non-null    float64
 15  women_55plus_rate  355 non-null    float64
dtypes: float64(15), object(1)


The only variable with missing values is women aged 55+

In [7]:
# Converting date variable to datetime
data['date'] = pd.to_datetime(data['date'])

In [8]:
# checking updated data types
data.dtypes

date                 datetime64[ns]
overall_rate                float64
men_rate                    float64
women_rate                  float64
men_16_19_rate              float64
women_16_19_rate            float64
men_20_24_rate              float64
women_20_24_rate            float64
men_25_34_rate              float64
women_25_34_rate            float64
men_35_44_rate              float64
women_35_44_rate            float64
men_45_54_rate              float64
women_45_54_rate            float64
men_55plus_rate             float64
women_55plus_rate           float64
dtype: object

In [9]:
# Checking for the dates where there are no entries for women at 55+
missing_dates = data[data['women_55plus_rate'].isna()]['date']
print("Dates with missing values:")
print(missing_dates)

Dates with missing values:
0     1948-01-01
1     1948-02-01
2     1948-03-01
3     1948-04-01
4     1948-05-01
         ...    
547   1993-08-01
548   1993-09-01
549   1993-10-01
550   1993-11-01
551   1993-12-01
Name: date, Length: 552, dtype: datetime64[ns]


With a total of 907 entries for each age range. There were 355 non null values and 552 entries with missing values for women 55+, which total 907.
We can conclude from this that there were not any entries recorded for women at 55+ between the years 1948 and 1993.
This column will not be removed due to its importance in later years.
One theory could be that women in this age group did not work, or the amount of workforce for that age group was very low that it did not get recorded.

In [10]:
data.describe().T

count       mean       std  min    25%   50%    75%   max
overall_rate       907.0   5.715877  1.707641  2.5   4.40   5.5   6.75  14.7
men_rate           907.0   5.585557  1.878606  2.3   4.30   5.3   6.70  13.5
women_rate         907.0   5.974642  1.580621  2.7   4.80   5.8   6.90  16.2
men_16_19_rate     907.0  16.809592  4.535748  6.4  14.10  16.6  19.10  30.7
women_16_19_rate   907.0  15.155017  3.662070  5.8  13.00  15.1  17.40  37.4
men_20_24_rate     907.0   9.759316  3.112013  3.2   7.90   9.3  11.30  23.1
women_20_24_rate   907.0   8.726240  2.442725  3.0   7.00   8.6  10.10  27.9
men_25_34_rate     907.0   5.201764  2.148678  1.5   3.70   4.9   6.40  14.1
women_25_34_rate   907.0   6.092944  1.578419  2.6   4.95   5.9   7.10  15.0
men_35_44_rate     907.0   3.934840  1.591589  1.1   2.80   3.7   4.80  10.5
women_35_44_rate   907.0   4.701985  1.262822  1.8   3.80   4.6   5.40  12.8
men_45_54_rate     907.0   3.765160  1.471712  1.2   2.80   3.5   4.50  11.3
women_45_54_rate   907.0   3.974862  1.213603  1.7   3.10   3.8   4.60  13.3
men_55plus_rate    907.0   3.921610  1.279589  1.5   3.00   3.7   4.50  12.1
women_55plus_rate  355.0   3.846197  1.509408  1.9   2.95   3.5   4.20  15.4

Using the describe method we can gain an understanding that younger people tend to have a higher unemployment rate, the rate decreases as they frown in age and at the age of 35 and older the unemployment rate seems to be constant.
As a mean it seems that men have a higher unemployment rate, but looking at the Max values women tend to have a higher unemployment rate. We could conclude from this that factor that cause increased unemployment rates affect women more than men.

Due to the comparison between men_45_54_rate and men_55plus_rate is also close while having no missing values, and the comparison between women_45_54_rate and women_55plus_rate is very close with women_55plus_rate having missing values.
We can fill in for the missing values for women_55plus_rate according to women_45_54_rate without disrupting the validity of the data

In [11]:
#  Function for filling missing values for women 55+ 
def fill_NA(row):
    if pd.isna(row['women_55plus_rate']):
        # reduction of 0.13 to each value is used as it is the difference between the mean values of women 45-54 and women 55+
        return row['women_45_54_rate'] - 0.13
    else:
        return row['women_55plus_rate']

# Applying the function
data['women_55plus_rate'] = data.apply(fill_NA, axis=1)

In [12]:
# There is a slight change in the values for women_55plus_rate but not enough to cause a disruption
data.describe().T

count       mean       std   min    25%    50%    75%   max
overall_rate       907.0   5.715877  1.707641  2.50   4.40   5.50   6.75  14.7
men_rate           907.0   5.585557  1.878606  2.30   4.30   5.30   6.70  13.5
women_rate         907.0   5.974642  1.580621  2.70   4.80   5.80   6.90  16.2
men_16_19_rate     907.0  16.809592  4.535748  6.40  14.10  16.60  19.10  30.7
women_16_19_rate   907.0  15.155017  3.662070  5.80  13.00  15.10  17.40  37.4
men_20_24_rate     907.0   9.759316  3.112013  3.20   7.90   9.30  11.30  23.1
women_20_24_rate   907.0   8.726240  2.442725  3.00   7.00   8.60  10.10  27.9
men_25_34_rate     907.0   5.201764  2.148678  1.50   3.70   4.90   6.40  14.1
women_25_34_rate   907.0   6.092944  1.578419  2.60   4.95   5.90   7.10  15.0
men_35_44_rate     907.0   3.934840  1.591589  1.10   2.80   3.70   4.80  10.5
women_35_44_rate   907.0   4.701985  1.262822  1.80   3.80   4.60   5.40  12.8
men_45_54_rate     907.0   3.765160  1.471712  1.20   2.80   3.50   4.50  11.3
women_45_54_rate   907.0   3.974862  1.213603  1.70   3.10   3.80   4.60  13.3
men_55plus_rate    907.0   3.921610  1.279589  1.50   3.00   3.70   4.50  12.1
women_55plus_rate  907.0   3.866968  1.212028  1.57   3.07   3.67   4.47  15.4

In [13]:
data.to_csv('Updated_sex_unemployment_rates.csv', index = False)